### PR TITLE
Add heatmap

### PR DESCRIPTION
**Pull request by devangcx**

### CODE REVIEW

**Comment by devangcx** [comment]
Hey @mostaphaRoudsari,

I have done minor adjustment to their module to create a Pandas Dataframe. This will make it easier for us to follow them and develop charts faster. We can introduce ladybug-pandas at a later stage. What do you think of this strategy? If you don't like the strategy, I will need to edit the Dataframe object coming out of laydbug-pandas as per [this](https://github.com/CenterForTheBuiltEnvironment/clima/blob/main/my_project/extract_df.py#L46-L273) function. My opinion is that we should do this switch later.

Here's a heatmap created from this arrangement.

newplot

File: ladybug_charts/_schema.py [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+            "#ff7b00",
+            "#fffc00",
+            "#ffff7b",
+            "#ffffff",
+        ],
+        "unit": radiation_unit,
+        "range": [0, 1200],
+    },
+    "hor_ir_rad": {
+        "name": "´Horizontal infrared radiation",
```
Comment (mostaphaRoudsari) [inline ladybug_charts/_schema.py]: I sent a PR to fix it on their side but we should also fix it here:

```suggestion
        "name": "Horizontal infrared radiation",
```

File: ladybug_charts/_schema.py [hunk 2]
```diff
@@ -0,0 +1,648 @@
+"""Schema to control visualization of Ladybug charts."""
```
Comment (mostaphaRoudsari) [inline ladybug_charts/_schema.py]: For files like this that we have copied and pasted the module from Clima's work, we should put a disclaimer on top with more information. Here is an example: 

https://github.com/ladybug-tools/honeybee/blob/3a8aacee5c98136d6393e33c2196b179a74b63e9/honeybee_plus/radiance/sky/gendaylit.py#L1-L14

File: ladybug_charts/to_df.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
+"""Create a Pandas DataFrame from an EPW file."""
```
Comment (mostaphaRoudsari) [inline ladybug_charts/to_df.py]: Similar to the above. Let's credit their project.

File: ladybug_charts/to_figure.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+from .to_df import DataPoint
+from math import ceil, floor
+
+
+def heatmap(df: Df, var: DataPoint, global_local: str = "global") -> Figure:
+    """Create a plotly heatmap figure.
+
+    Args:
+        df: A pandas dataframe.
+        var: A string with the name of the variable to be plotted. Choose from one of the
```
Comment (mostaphaRoudsari) [inline ladybug_charts/to_figure.py]: This is incomplete and also seems to be wrong. You're expecting a DataPoint and not a string.

File: ladybug_charts/to_figure.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        global_local: A string with the name of the global or local time.
```
Comment (mostaphaRoudsari) [inline ladybug_charts/to_figure.py]: What is global vs local time here? I think you need a bit more information to make it easier for people to use it.

File: requirements.txt [hunk 6]
```diff
@@ -1 +1,4 @@
-ladybug-core==0.39.25
+plotly==5.4.0
+pandas==1.3.5
+pvlib==0.9.0
+pythermalcomfort==1.10.0
```
Comment (mostaphaRoudsari) [inline requirements.txt:4]: Do we need `pvlib` and `pythermalcomfort` here? I think we should use `ladybug-core` and `ladybug-comfort` instead.

File: tests/to_figure_test.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+from ladybug_charts.to_figure import heatmap
+from ladybug_charts.to_df import create_df, DataPoint
+from plotly.graph_objects import Figure
+
+
+def test_heatmap():
+    epw_path = r'tests/assets/weather/boston.epw'
+    df = create_df(epw_path)[0]
```
Comment (mostaphaRoudsari) [inline tests/to_figure_test.py]: Really nice to see how easy the process is. For the sake of clarity for DF as dataframe vs DF as Dragonfly vs DF as Daylight Factor, I suggest changing the name to `create_data_frame` or even better - inject it to EPW class and make it `to_dataframe`.

```suggestion
    df = create_dataframe(epw_path)[0]
```

Preferred approach in my mind:

```python
epw = EPW(epw_path )
epw_df = epw.to_dataframe()

## or even

epw = EPW(epw_path )
figure = epw.relative_humidity.heatmap()

```

**Review by mostaphaRoudsari — COMMENTED** [review]
Thank you @devngc! I agreed with your suggestion to keep using the Datafram from Clima for the first iteration. This should help us to get a working prototype out sooner which is very important. I left some comments about credits, naming and API design.

File: requirements.txt (hunk 6 above)
Comment (devangcx) [reply requirements.txt:4]: https://github.com/ladybug-tools/ladybug-charts/issues/13

File: ladybug_charts/to_figure.py (hunk 5 above)
Comment (devangcx) [reply ladybug_charts/to_figure.py]: I had not looked into it. I have updated it to allow a custom range for the data.

**Comment by devangcx** [comment]
After thinking more about this. I think I will make the transition to using ladybug-pandas as a next step before going to work on monthly chart. That should also remove dependency on pvlib and pythermalcomfort.

**Comment by devangcx** [comment]
https://github.com/ladybug-tools/ladybug-charts/pull/14

**Pull request closed without merging**